x-pack/plugins/observability_solution/infra/server/routes/custom_dashboards/save_custom_dashboard.ts

```diff
@@ -37,10 +40,18 @@ export function initSaveCustomDashboardRoute(framework: KibanaFramework) {
 
       await checkCustomDashboardsEnabled(uiSettingsClient);
 
-      const payload: InfraSaveCustomDashboardsRequestPayload = request.body;
+      const payload: InfraSaveCustomDashboardsResponseBody = {
+        ...request.body,
+        assetType: request.params.assetType,
+      };
       const customDashboards = await findCustomDashboard(payload.assetType, savedObjectsClient);
 
-      if (customDashboards.total === 0) {
+      const existingCustomDashboard = await findCustomDashboardByDashboardId(
```

Comment:
Suggestion: This works fine, I would just propose the following:

- Change the payload to receive `{ id?: number // custom dashboard saved object id, dashboardSavedObjectId: number, dashboardFilterAssetIdEnabled: boolean }`
- if `id` is not undefined, call `create`, otherwise `update`. [similar to APM](https://github.com/elastic/kibana/blob/main/x-pack/plugins/observability_solution/apm/server/routes/custom_dashboards/save_service_dashboard.ts#L29)

The `findCustomDashboardByDashboardId` has the benefit of validating whether the `dashboardSavedObjectId` exists before creating or updating the object, but its main purpose seems to be in retrieving the custom dashboard saved object id so that this id can be used in the create and update operations.

Validating whether the `dashboardSavedObjectId` really exists in a saved object is an interesting validation we can do - if we want to make sure the API is not trying to link an inexisting dashboard.

Reply:
Thanks for the comment! I tried that and pushed the changes - what I see as a downside is that the `id` needs to be passed to the payload in order to update the SO and also if it's not passed a new object will be created - which can be an issue as we don't want to create a new object with the same id (in case the id is not passed there will be duplicates)
I am thinking about how we can improve that and I know that checking if the object exists is not ideal in general but it can be used to validate and to not allow posting the same `dashboardSavedObjectId` (the frontend should not do that but still) and create a PUT endpoint to be called in this case we want to update the object (thinking out loud here)